CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -155,16 +155,26 @@ const char *_CFProcessPath(void) {
 }
 #endif
 
-#if DEPLOYMENT_TARGET_LINUX
+#if DEPLOYMENT_TARGET_LINUX || DEPLOYMENT_TARGET_FREEBSD
 #include <unistd.h>
 #if __has_include(<syscall.h>)
 #include <syscall.h>
 #else
 #include <sys/syscall.h>
 #endif
 
+#if DEPLOYMENT_TARGET_FREEBSD
+#include <pthread_np.h>             // for _CFIsMainThread
+#endif
+
 Boolean _CFIsMainThread(void) {
+#if DEPLOYMENT_TARGET_LINUX
     return syscall(SYS_gettid) == getpid();
+#elif DEPLOYMENT_TARGET_FREEBSD
+	return pthread_main_np() != 0;
```

Comment:
There are whitespace mismatches — make sure your editor matches the settings of the file around it?